projects/common/src/telemetry/providers/freshpaint/freshpaint-provider.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { TelemetryProviderConfig, UserTelemetryProvider, UserTraits } from '../../telemetry';
+import { Dictionary } from './../../../utilities/types/types';
+import { FreshPaint, loadFreshPaint } from './load-snippet';
+
+@Injectable({ providedIn: 'root' })
+export class FreshPaintTelemetry<InitConfig extends TelemetryProviderConfig>
+  implements UserTelemetryProvider<InitConfig> {
+  private freshPaint?: FreshPaint;
+
+  public initialize(config: InitConfig): void {
+    this.freshPaint = loadFreshPaint();
```

Comment:
btw, the script loads one global object. So, we can't support multiple configs for the same telemetry provider.

Reply:
That's fine - can only do it if they support it. Now that this PR is getting more mature, can we start considering separating it down - I'd think the telemetry orchestration, tracking directive, and each specific provider can all be separate PRs.

Reply:
Sure. let's take it to 90ish %. Splitting it up would be pretty easy